### PR TITLE
fix: Prevent query parameters from being optional when header parameters are required.

### PR DESCRIPTION
**Pull request by shibeshduw**
With this change, the file serializer in `openapi-generator` will always ensure that query parameters are not marked optional when header parameters are required by conditionally passing the correct optional flag to `serializeParamsForSignature`.

Closes SAP/cloud-sdk-js#4928.

### CODE REVIEW

**Comment by deekshas8** [comment]
@shibeshduw I am not sure if I am misunderstanding, but I understood the user's issue as: incorrect code is generated when there are optional query and required header params, I thought this was something we covered in our [tests](https://github.com/SAP/cloud-sdk-js/blob/5847e0af7e04286e58a6794bc297dffe5895430d/packages/openapi-generator/src/file-serializer/operation.spec.ts#L195). 

Is this not the case?

**Comment by shibeshduw** [comment]
Yes you're correct. But the manner in which we were [checking for optional modifier](https://github.com/SAP/cloud-sdk-js/blob/5847e0af7e04286e58a6794bc297dffe5895430d/packages/openapi-generator/src/file-serializer/operation.ts#L47) was wrong. I was suggesting that we make the order of parameters dynamic but after speaking with @marikaner , I go to know that this would be a breaking change. So now I have reverted the changes and added a minimal fix for the issue.

File: packages/openapi-generator/src/file-serializer/operation.ts [hunk 1]
```diff
@@ -36,15 +36,24 @@ ${operation.operationId}: (${serializeOperationSignature(
...
-    'headerParameters'
+    'headerParameters',
+    allOptionalHeaders
   );
 
   const queryParams = serializeParamsForSignature(
     operation,
     'queryParameters',
-    headerParams?.includes('?')
+    allOptionalHeaders ? allOptionalQuery : false
```
Comment (marikaner) [inline packages/openapi-generator/src/file-serializer/operation.ts:56]:
```suggestion
    allOptionalHeaders && allOptionalQuery
```

**Review by marikaner — APPROVED** [review]
LGTM

**Comment by marikaner** [comment]
Please still correct the title 😊

**Pull request merged**